src/QMCWaveFunctions/BsplineFactory/SplineR2R.h

```diff
@@ -71,16 +72,33 @@ class SplineR2R : public BsplineSet
   hContainer_type myH;
   ghContainer_type mygH;
 
+  IndexType m_true_norbs;
+
 public:
   SplineR2R()
   {
-    is_complex = false;
-    className  = "SplineR2R";
-    KeyWord    = "SplineR2R";
+    is_complex   = false;
+    className    = "SplineR2R";
+    KeyWord      = "SplineR2R";
+    m_true_norbs = 0;
   }
 
+  // Sets the number of spline SPOs
+  // Note that this might be < number of splines due to padding
+  void setOrbitalSetSize(int norbs) override { m_true_norbs = norbs; }
+
+  // Gives the number of splined SPOs
+  int getOrbitalSetSize() const override { return m_true_norbs; }
+
+  // Gives the number of splined SPOs
+  // JPT DEBUG: Shouldn't this be different from orbitalsetsize?
+  int getBasisSetSize() const override { return SplineInst->num_splines(); }
```

Comment:
Spline basisset size is (coefs_size/norb_padded). This number should not be known outside SPOSet.
The unique thing about spline coefficients is it is stored (nbasis, norb) while LCAO stores (norb, nbasis).
Indeed, getBasisSetSize should only be used when constructing SPOSet object.

Reply:
I mixed up BasisSetSize/OrbitalSetSize. Fixed now to be consistent. 
BasisSetSize = Number of points per spline
OrbitalSetSize = Noccupied orbs
Nsplines = Padded number of splines according to underlying spline class.